### PR TITLE
CI: Setup vcpkg caching on GHA

### PR DESCRIPTION
**Pull request by MehdiChinoune**
See: https://github.com/microsoft/vcpkg/blob/master/docs/users/binarycaching.md#github-packages

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -138,6 +139,14 @@ jobs:
         # note: system path must be modified in a previous step to it's use
         echo "$Env:VCPKG_INSTALLATION_ROOT\installed\${{ matrix.config.vc }}\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
         echo "${{github.workspace}}\build\openvdb\openvdb\${{ matrix.config.build }}" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+    - name: Setup NuGet Credentials
+      run: |
+        & $(vcpkg fetch nuget | select -last 1) sources add `
+        -source "https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json" `
+        -storepasswordincleartext `
```
Comment (Idclip) [inline .github/workflows/build.yml:146]: what is `-storepasswordincleartext` doing here?

Comment (Idclip) [inline .github/workflows/build.yml:146]: what is `-storepasswordincleartext` doing here?

Comment (MehdiChinoune) [reply .github/workflows/build.yml:146]: I don't know I have just copied/pasted the recommended setup from the link mentioned above

Comment (Idclip) [reply .github/workflows/build.yml:146]: Thanks, will investigate but might take some time

Comment (danrbailey) [reply .github/workflows/build.yml:146]: Not an expert on this, but after a quick look, I believe this flag only needs to be set on Linux/MacOS as dotnet only supports encryption on Windows.

Comment (MehdiChinoune) [reply .github/workflows/build.yml]: Removed!
Although the password stored is `"${{ secrets.GITHUB_TOKEN }}"`, nothing private is passed/stored.

**Pull request closed without merging**